Extensions/PathfindingBehavior/tests/legacypathfindingruntimebehavior.spec.js

```diff
@@ -0,0 +1,689 @@
+        targetY = playerY + 2 * cellSize;
+        break;
+      case 'down':
+        obstacleX = playerX;
+        obstacleY =
+          playerY +
+          playerBottomBorder +
+          obstacleTopBorder -
+          cellSize +
+          inclusiveMargin;
+        targetX = playerX;
+        targetY = playerY - 2 * cellSize;
+        break;
+    }
+
+    obstacle.setPosition(obstacleX, obstacleY);
+    // so obstacles register
+    runtimeScene.renderAndStep(1000 / 60);
+
+    // move away from the obstacle
+    player.setPosition(playerX, playerY);
+    player.getBehavior(pathFindingName).moveTo(runtimeScene, targetX, targetY);
+    //expect(player._animationFrame).to.be(null);
+    expect(player.getBehavior(pathFindingName).pathFound()).to.be(true);
+    expect(player.getBehavior(pathFindingName).getNodeCount()).to.be(3);
+  });
+
+  it("mustn't find a path when an obstacle is slightly overlaping the only first cell to go", function () {
+    let obstacleX;
+    let obstacleY;
+    let targetX;
+    let targetY;
+    switch (direction) {
+      case 'right':
+        // right on the player right
+        obstacleX =
+          playerX +
+          playerRightBorder +
+          obstacleLeftBorder -
+          cellSize +
+          inclusiveMargin -
+          1;
+        obstacleY = playerY;
+        targetX = playerX - 2 * cellSize;
+        targetY = playerY;
+        break;
+      case 'left':
+        obstacleX =
+          playerX -
+          playerLeftBorder -
+          obstacleRightBorder +
+          cellSize -
+          inclusiveMargin +
+          1;
+        obstacleY = playerY;
+        targetX = playerX + 2 * cellSize;
+        targetY = playerY;
+        break;
+      case 'up':
+        obstacleX = playerX;
+        obstacleY =
+          playerY -
+          playerTopBorder -
+          obstacleBottomBorder +
+          cellSize -
+          inclusiveMargin +
+          1;
+        targetX = playerX;
+        targetY = playerY + 2 * cellSize;
+        break;
+      case 'down':
+        obstacleX = playerX;
+        obstacleY =
+          playerY +
+          playerBottomBorder +
+          obstacleTopBorder -
+          cellSize +
+          inclusiveMargin -
+          1;
+        targetX = playerX;
+        targetY = playerY - 2 * cellSize;
+        break;
+    }
+
+    obstacle.setPosition(obstacleX, obstacleY);
+    // so obstacles register
+    runtimeScene.renderAndStep(1000 / 60);
+
+    // move away from the obstacle
+    player.setPosition(playerX, playerY);
+    player.getBehavior(pathFindingName).moveTo(runtimeScene, targetX, targetY);
+    if (cellSize == 160) {
+      // The obstacle is the size of the cell so the player can bypass it.
+      expect(player.getBehavior(pathFindingName).getNodeCount()).to.be.above(3);
+    } else {
+      expect(player.getBehavior(pathFindingName).pathFound()).to.be(false);
+    }
+  });
+
+  it('can find a path with an obstacle adjascent to the target', function () {
```

Comment:
typo: adjacent

Reply:
Yes, I'm really sorry for all the typo. I'll try another IDE because not much is working on Eclipse for JS.

Reply:
Definitely give a try at VS Code (great integration with typescript/javascript). You can also try WebStorm if you want something a bit more "heavy", but I like the simplicity of VS Code + the great editing capabilities (multi selection) and the support for TS/JS is very good.